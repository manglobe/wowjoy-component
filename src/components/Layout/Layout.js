import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
const Wrap = styled.div`
  width: 100%;
  min-height: 100vh;
  position: relative;
  overflow: hidden;
  & > header {
    height: 64px;
    line-height: 64px;
    width: 100%;
  }
  & > footer {
    height: 64px;
    line-height: 64px;
    width: 100%;
  }
  ${props => props.defaultStyles};
`;

const AsideLeft = styled.aside``;
const Main = styled.main``;
const Center = styled.article``;
const AsideRight = styled.aside``;
class Layout extends PureComponent {
  render() {
    const {
      className,
      defaultStyles,
      header,
      asideLeft,
      children,
      asideRight,
      footer
    } = this.props;
    return (
      <Wrap
        defaultStyles={defaultStyles}
        className={"wjc-layout " + className || ""}
      >
        {header && <header className={"wjc-layout-header"}>{header}</header>}
        <Main>
          {asideLeft && (
            <AsideLeft className={"wjc-layout-aside__left"}>
              {asideLeft}
            </AsideLeft>
          )}
          {children && (
            <Center className={"wjc-layout-center"}>{children}</Center>
          )}
          {asideRight && (
            <AsideRight className={"wjc-layout-aside__right"}>
              {asideRight}
            </AsideRight>
          )}
        </Main>
        {footer && <footer className={"wjc-layout-footer"}>{footer}</footer>}
      </Wrap>
    );
  }
}

Layout.propTypes = {
  className: PropTypes.string,
  defaultStyles: PropTypes.string,
  header: PropTypes.node,
  asideLeft: PropTypes.node,
  children: PropTypes.node,
  asideRight: PropTypes.node,
  footer: PropTypes.node
};
export default Layout;
