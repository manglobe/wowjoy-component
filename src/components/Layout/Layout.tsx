import * as React from "react";
import styled from "styled-components";
const Wrap = styled.div`
  width: 100%;
  min-height: 100vh;
  position: relative;
  overflow: hidden;
  & > header {
    height: 64px;
    line-height: 64px;
    width: 100%;
  }
  & > footer {
    height: 64px;
    line-height: 64px;
    width: 100%;
  }
  ${(props: { defaultStyles?: string }) => props.defaultStyles};
`;

const AsideLeft = styled.aside``;
const Main = styled.main``;
const Center = styled.article``;
const AsideRight = styled.aside``;

export interface Props {
  className?: string;
  defaultStyles?: string;
  header?: React.ReactNode;
  asideLeft?: React.ReactNode;
  children?: React.ReactNode;
  asideRight?: React.ReactNode;
  footer?: React.ReactNode;
}
class Layout extends React.PureComponent<Props, {}> {
  render() {
    const {
      className,
      defaultStyles,
      header,
      asideLeft,
      children,
      asideRight,
      footer
    } = this.props;
    return (
      <Wrap
        defaultStyles={defaultStyles}
        className={"wjc-layout " + className || ""}
      >
        {header && <header className={"wjc-layout-header"}>{header}</header>}
        <Main>
          {asideLeft && (
            <AsideLeft className={"wjc-layout-aside__left"}>
              {asideLeft}
            </AsideLeft>
          )}
          {children && (
            <Center className={"wjc-layout-center"}>{children}</Center>
          )}
          {asideRight && (
            <AsideRight className={"wjc-layout-aside__right"}>
              {asideRight}
            </AsideRight>
          )}
        </Main>
        {footer && <footer className={"wjc-layout-footer"}>{footer}</footer>}
      </Wrap>
    );
  }
}

export default Layout;
