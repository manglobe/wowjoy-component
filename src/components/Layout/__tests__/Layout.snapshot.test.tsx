import * as React from "react";
import Layout from "../index";
import * as renderer from "react-test-renderer";
import * as TestUtils from "react-dom/test-utils";
import { MemoryRouter, Route } from "react-router-dom";
import { mount, configure } from "enzyme";
import * as ReactSixteenAdapter from "enzyme-adapter-react-16";
const adapter = ReactSixteenAdapter as any;
configure({ adapter: new adapter.default() });

const { renderIntoDocument } = TestUtils;
const rendersCorrectly = (propName: string) => {
  it(`renders ${propName} correctly`, () => {
    const reactObj = <Layout {...{ [propName]: propName + " content" }} />;
    const tree = renderer.create(reactObj).toJSON();
    expect(tree).toMatchSnapshot();
  });
};
describe("Layout", () => {
  const reactObj = <Layout>Layout Content</Layout>;
  it("renders correctly", () => {
    const tree = renderer.create(reactObj).toJSON();
    expect(tree).toMatchSnapshot();
  });
  ["header", "asideLeft", "children", "asideRIght", "footer"].forEach(ele => {
    rendersCorrectly(ele);
  });
});
