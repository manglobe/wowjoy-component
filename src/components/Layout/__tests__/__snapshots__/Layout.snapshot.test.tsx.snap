// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders asideLeft correctly 1`] = `
<div
  className="wjc-layout undefined sc-bdVaJa dgEXNJ"
>
  <main
    className="sc-htpNat KtFsv"
  >
    <aside
      className="wjc-layout-aside__left sc-bwzfXH uTxCW"
    >
      asideLeft content
    </aside>
  </main>
</div>
`;

exports[`Layout renders asideRIght correctly 1`] = `
<div
  className="wjc-layout undefined sc-bdVaJa dgEXNJ"
>
  <main
    className="sc-htpNat KtFsv"
  />
</div>
`;

exports[`Layout renders children correctly 1`] = `
<div
  className="wjc-layout undefined sc-bdVaJa dgEXNJ"
>
  <main
    className="sc-htpNat KtFsv"
  >
    <article
      className="wjc-layout-center sc-bxivhb bUZGRr"
    >
      children content
    </article>
  </main>
</div>
`;

exports[`Layout renders correctly 1`] = `
<div
  className="wjc-layout undefined sc-bdVaJa dgEXNJ"
>
  <main
    className="sc-htpNat KtFsv"
  >
    <article
      className="wjc-layout-center sc-bxivhb bUZGRr"
    >
      Layout Content
    </article>
  </main>
</div>
`;

exports[`Layout renders footer correctly 1`] = `
<div
  className="wjc-layout undefined sc-bdVaJa dgEXNJ"
>
  <main
    className="sc-htpNat KtFsv"
  />
  <footer
    className="wjc-layout-footer"
  >
    footer content
  </footer>
</div>
`;

exports[`Layout renders header correctly 1`] = `
<div
  className="wjc-layout undefined sc-bdVaJa dgEXNJ"
>
  <header
    className="wjc-layout-header"
  >
    header content
  </header>
  <main
    className="sc-htpNat KtFsv"
  />
</div>
`;
